packages/flutter/lib/src/rendering/child_view.dart

```diff
@@ -0,0 +1,207 @@
+  // before we get removed from the old render object. By the time we get around
+  // to computing our layout, we must be back to just having one render object.
+  int _attachments = 0;
+  bool get _attached => _attachments > 0;
+
+  void _attach() {
+    assert(_attachments >= 0);
+    ++_attachments;
+    if (_viewToken != null && _viewKey == null)
+      _addChildToViewHost();
+  }
+
+  void _detach() {
+    assert(_attached);
+    --_attachments;
+    scheduleMicrotask(_removeChildFromViewHostIfNeeded);
+  }
+
+  void _removeChildFromViewHostIfNeeded() {
+    assert(_attachments >= 0);
+    if (_attachments == 0)
+      _removeChildFromViewHost();
+  }
+
+  Future<mojom.ViewLayoutInfo> _layout({ Size size, double scale }) async {
+    assert(_attached);
+    assert(_attachments == 1);
+    assert(_viewKey != null);
+    int width = (size.width * scale).round();
+    int height = (size.height * scale).round();
+    mojom.BoxConstraints childConstraints = new mojom.BoxConstraints()
+      ..minWidth = width
+      ..maxWidth = width
+      ..minHeight = height
+      ..maxHeight = height;
+    mojom.ViewLayoutParams layoutParams = new mojom.ViewLayoutParams()
+      ..constraints = childConstraints
+      ..devicePixelRatio = scale;
+    return (await _viewHost.layoutChild(_viewKey, layoutParams)).info;
+  }
+
+  String toString() {
+    return '$runtimeType(url: $url)';
+  }
+}
+
+class RenderChildView extends RenderBox {
+  RenderChildView({
+    ChildViewConnection child,
+    double scale
+  }) : _child = child, _scale = scale;
+
+  ChildViewConnection get child => _child;
+  ChildViewConnection _child;
+  void set child (ChildViewConnection value) {
+    if (value == _child)
+      return;
+    if (attached)
+      _child?._detach();
+    _child = value;
+    _layoutInfo = null;
+    if (attached)
+      _child._attach();
+
+    if (_child._viewToken != null) {
+      // We've already connected to the view, so we're ready to invalidate our
+      // layout immediately.
+      markNeedsLayout();
+    } else {
+      // Otherwise, we're still in the process of connecting, so we need to
+      // repaint now (to remove any old child view), and we need to watch for
+      // the view token resolving before attempting layout.
+      markNeedsPaint();
+      _child._unresolvedViewToken.then(_handleViewTokenResolved);
+    }
+  }
+
+  double get scale => _scale;
+  double _scale;
+  void set scale (double value) {
+    if (value == _scale)
+      return;
+    _scale = value;
+    markNeedsLayout();
+  }
+
+  bool get sizedByParent => true;
+  bool get alwaysNeedsCompositing => true;
+
+  void attach() {
+    super.attach();
+    _child?._attach();
+  }
+
+  void detach() {
+    _child?._detach();
+    super.detach();
+  }
+
+  void performResize() {
+    size = constraints.biggest;
+  }
+
+  void performLayout() {
+    if (_child != null)
+      _child._layout(size: size, scale: scale).then(_handleLayoutInfoChanged);
```

Comment:
this hurts my head. I guess really what's going on here is you're using our layout needed signal to know when the size changes, since you only need to call _child._layout when the size changes. That seems reasonable.